ibm/service/contextbasedrestrictions/resource_ibm_cbr_zone.go

```diff
@@ -344,88 +309,95 @@ func resourceIBMCbrZoneCreate(context context.Context, d *schema.ResourceData, m
 		return diag.FromErr(fmt.Errorf("CreateZoneWithContext failed %s\n%s", err, response))
 	}
 
-	// handle Eventual consistency case
-	err = readNewCbrZone(contextBasedRestrictionsClient, context, *zone.ID)
-	if err != nil {
-		return diag.FromErr(err)
-	}
-
 	d.SetId(*zone.ID)
-	return resourceIBMCbrZoneRead(context, d, meta)
-}
-
-func resourceIBMCbrZoneRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	contextBasedRestrictionsClient, err := meta.(conns.ClientSession).ContextBasedRestrictionsV1()
-	if err != nil {
-		return diag.FromErr(err)
-	}
 
-	var zone *contextbasedrestrictionsv1.Zone
-	var version string
-	var found bool
-	zone, version, found, err = getZone(contextBasedRestrictionsClient, context, d.Id())
-	if err != nil {
-		return diag.FromErr(err)
-	}
-	if !found {
-		d.SetId("")
-		return nil
+	version := response.Headers.Get("Etag")
+	if err := ResourceIBMCbrZoneSetData(zone, version, d); err != nil {
```

Comment:
Let's make `ResourceIBMCbrZoneSetData ` consistent with `resourceIBMCbrRuleSetData` -- accepting the `response` object rather than the `version` directly.

Reply:
Updated